Mathlib/Analysis/Calculus/BumpFunction/SmoothApproxLp.lean

```diff
@@ -0,0 +1,106 @@
+/-
+Copyright (c) 2025 Moritz Doll. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Moritz Doll
+-/
+import Mathlib.MeasureTheory.Function.ContinuousMapDense
+import Mathlib.MeasureTheory.Function.UniformIntegrable
+import Mathlib.Geometry.Manifold.SmoothApprox
+import Mathlib.Tactic.MoveAdd
+
+/-!
+
+# Density of smooth compactly supported functions in `Lp`
+
+In this file, we prove that `Lp` functions can be approximated by smooth compactly supported
+functions for `p < ∞`.
+
+This result is recorded in `MeasureTheory.MemLp.exist_sub_eLpNorm_le`.
+-/
+
+
+variable {α β E F : Type*} [MeasurableSpace E] [NormedAddCommGroup F]
+
+open scoped Nat NNReal ContDiff
+open MeasureTheory Pointwise ENNReal
+
+theorem MeasureTheory.eLpNorm_sub_le_of_dist_bdd' (μ : Measure E := by volume_tac)
+    {p : ℝ≥0∞} (hp : p ≠ ⊤) {c : ℝ} (hc : 0 ≤ c) {f g : E → F} {s : Set E}
+    (h : ∀ x, dist (f x) (g x) ≤ c) (hs : MeasurableSet s) (hs₁ : f.support ⊆ s)
+    (hs₂ : g.support ⊆ s) :
+    eLpNorm (f - g) p μ ≤ ENNReal.ofReal c * μ s ^ (1 / p.toReal) := by
+  have hs₃ : s.indicator (f - g) = f - g := by
+    rw [Set.indicator_eq_self]
+    exact (Function.support_sub _ _).trans (Set.union_subset hs₁ hs₂)
+  rw [← hs₃]
+  exact eLpNorm_indicator_sub_le_of_dist_bdd μ hp hs hc (fun x _ ↦ h x)
```

Comment:
Would you mind moving this, as well as the existing result in `UniformIntegrable.lean`, closer to [MeasureTheory.eLpNorm_indicator_const](https://leanprover-community.github.io/mathlib4_docs/Mathlib/MeasureTheory/Function/LpSeminorm/Basic.html#MeasureTheory.eLpNorm_indicator_const)? Or is it not possible for some reason?